pkg/api/norman/customization/aks/handler.go

```diff
@@ -177,20 +178,25 @@ func (h *handler) checkCredentials(req *http.Request) (int, error) {
 	if cred.ClientSecret == "" {
 		return http.StatusBadRequest, fmt.Errorf("must provide clientSecret")
 	}
+
+	clientEnvironment := ""
+	if cred.Environment != "" {
+		clientEnvironment = cred.Environment
+	}
+	azureEnvironment := GetEnvironment(clientEnvironment)
+
+	cred.BaseURL = azureEnvironment.ResourceManagerEndpoint
+	cred.AuthBaseURL = azureEnvironment.ActiveDirectoryEndpoint
+
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 	defer cancel()
 	if cred.TenantID == "" {
-		cred.TenantID, err = azureutil.FindTenantID(ctx, azure.PublicCloud, cred.SubscriptionID)
+		cred.TenantID, err = azureutil.FindTenantID(ctx, azureEnvironment, cred.SubscriptionID)
 		if err != nil {
 			return http.StatusBadRequest, fmt.Errorf("could not find tenant ID: %w", err)
```

Comment:
Suggestion

`return http.StatusBadRequest, fmt.Errorf("could not find tenant ID for Azure environment %w: %w", azureEnvironment, err)`

Reply:
great suggestion, thank you